app/Http/Controllers/ApplicationController.php

```diff
@@ -175,6 +186,36 @@ public function update($id, Request $request)
         unset($application->test_type);
         unset($application->test_score);
     }
+
+    // If there is not already a file, just throw the name in the uploads column
+      $upload = $request->file('upload');
+      if ($request->hasFile('upload') && empty($application->upload)) {
+        $filename = $upload->getClientOriginalName();
+        $upload->move(base_path('/storage/app/uploads/'.$application->user_id.'/'), $filename);
```

Comment:
These two branches are so similar! It'd be nice to move the first two duplicated lines out into a shared `if ($request->hasFile('upload') { ... }` case rather than doing it separately (and then have a nested conditional for how we save it).